include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -54,14 +54,34 @@
         but OpenMP headers are not found or the compiler does not support OpenMP"
 #endif
 
-#if (defined(SYCL_LANGUAGE_VERSION) || defined(CL_SYCL_LANGUAGE_VERSION)) &&                                           \
-    (__has_include(<sycl/sycl.hpp>) || __has_include(<CL/sycl.hpp>))
+// Preliminary check SYCL availability
+#if (__has_include(<sycl/sycl.hpp>) || __has_include(<CL/sycl.hpp>))
+#    define _ONEDPL_SYCL_HEADER_PRESENT 1
+#endif
+#if defined(CL_SYCL_LANGUAGE_VERSION) || defined(SYCL_LANGUAGE_VERSION)
+#    define _ONEDPL_SYCL_LANGUAGE_VERSION_PRESENT 1
+#endif
+#if _ONEDPL_SYCL_HEADER_PRESENT && _ONEDPL_SYCL_LANGUAGE_VERSION_PRESENT
 #    define _ONEDPL_SYCL_AVAILABLE 1
 #endif
-#if ONEDPL_USE_DPCPP_BACKEND && !_ONEDPL_SYCL_AVAILABLE
+// Intel(R) oneAPI DPC++/C++ Compiler provides SYCL_LANGUAGE_VERSION without sycl.hpp inclusion
+#if _ONEDPL_SYCL_HEADER_PRESENT && !_ONEDPL_SYCL_LANGUAGE_VERSION_PRESENT && !defined(__INTEL_LLVM_COMPILER)
+#    define _ONEDPL_SYCL_POSSIBLY_AVAILABLE 1
+#endif
```

Comment:
The code is correct, however that's how I would do it:
```
#if _ONEDPL_SYCL_HEADER_PRESENT
#  if _ONEDPL_SYCL_LANGUAGE_VERSION_PRESENT
#    define _ONEDPL_SYCL_AVAILABLE 1
// Intel(R) oneAPI DPC++/C++ Compiler pre-defines SYCL_LANGUAGE_VERSION with -fsycl option
#  elif !defined(__INTEL_LLVM_COMPILER)
// Other compilers might define the macro in the SYCL header
#    define _ONEDPL_SYCL_POSSIBLY_AVAILABLE 1
#  endif
#endif
```

Reply:
Applied